### PR TITLE
Fish integration: Some autocompletions

### PR DESCRIPTION
**Pull request by jaimergp**
This PR implements basic autocompletions for Fish shell. It is able to suggest:

* `conda` subcommands
* existing environments for `conda activate`
* installed packages in current environment for `conda remove` and `conda upgrade/remove`

### CODE REVIEW

**Comment by kalefranz** [comment]
@jaimergp could you rebase against master? I tried for you, but I don't have the fish knowledge to have confidence in resolving the merge conflicts.

File: shell/conda.fish [hunk 1]
```diff
@@ -190,14 +190,13 @@ function activate --description 'Activate a conda environment.'
         else
           set -gx CONDA_DEFAULT_ENV $argv[1]
         end
-
         # Always store the full prefix path as CONDA_PREFIX
         set -gx CONDA_PREFIX (echo $PATH[1] | sed 's|/bin$||g')
 
         # check if there are any *.fish scripts in activate.d
         set -l activate_d $CONDA_PREFIX/etc/conda/activate.d
```
Comment (jaimergp) [inline shell/conda.fish]: Is this the correct place? In the root env? Or does every env have its own `etc/conda/(de)activate.d` folders? Should we take that possibility into account?

File: setup.py [hunk 2]
```diff
@@ -36,6 +36,7 @@
 
 scripts = ['shell/activate',
            'shell/deactivate',
+           'shell/conda.fish'
```
Comment (kalefranz) [inline setup.py]: Nah, `setup.py` can't really handle it correctly.  `scripts` would put `conda.fish` in the `./bin` directory, which is not quite where we want it.  Conda's own conda-build recipe puts `shell/conda.fish` at `$PREFIX/etc/fish/conf.d/conda.fish`, which I think is the "correct" place for it to go.

https://github.com/conda/conda/blob/master/conda.recipe/build.sh#L10-L11

File: shell/conda.fish (hunk 1 above)
Comment (jaimergp) [inline shell/conda.fish]: Ah, ok, got it now! Thanks.

File: setup.py (hunk 2 above)
Comment (jaimergp) [inline setup.py]: By the way, would it be possible to patch `config.fish` the same way the installer offers to patch `.bashrc`? Or, at least, give instructions during the installation process.

**Comment by jaimergp** [comment]
_(I think I posted this already, but can't seem to find it!)_

@kalefranz I think we're done here! I am wondering, though... could we offer to patch `config.fish` for fish users during the installation the same way `.bashrc` gets patched for bash users? Where should I look for that piece of code?

**Comment by kalefranz** [comment]
We're actually not patching .bashrc, as far as I know.

At least, I know the ".sh" installers don't do it. I don't think the native OS X or Windows installers do either, but I could be wrong. 

There's a large and important PR (https://github.com/conda/conda/pull/3175) that's going into 4.3 that unifies and cleans up bourne and c shells. After that settles, a few of the next steps will be ideas that conda.fish has already implemented. That is, in several ways, conda.fish is ahead. 

However things evolve, I'll make sure you're included in the discussion, and that fish is along for the ride :)

**Comment by jaimergp** [comment]
Last time I installed Miniconda on a coworker's PC I was offered the possibility to patch `.bashrc` so it includes `$CONDA/bin` in `$PATH`. I think it's the last step of the .sh wizard.

Also, should we take Fish into account in this [piece of code](https://github.com/conda/conda/blob/master/conda/cli/install.py#L80-L104) so it prints `conda activate env` instead of `source activate env`? I don't know if that has to do something with PR #3175, but just asking in case :)

**Comment by kalefranz** [comment]
> Also, should we take Fish into account in this piece of code...

If I didn't have plans to make this part of all the other shells more like fish, I'd say yes.  But I'm hopeful in the near future the instructions for everyone will be `conda activate env`.

**Pull request merged**